returnn/datasets/audio.py

```diff
@@ -402,15 +400,47 @@ def get_total_num_seqs(self, *, fast: bool = False) -> int:
         self._lazy_init()
         return len(self._data)
 
-    def get_data_shape(self, key):
+    def get_data_dtype(self, key: str) -> str:
+        """:return: dtype of data entry with `key`"""
+        if key == "data":
+            return "float32"
+        elif key == "classes":
+            return "int32"
+        elif key == "raw":
+            return "string"
+        elif key == "orth":
+            return "uint8"
+        else:
+            raise ValueError(f"{self}: unknown data key: {key}")
+
+    def get_data_keys(self) -> List[str]:
+        """:return: available data keys"""
+        keys = []
+        if self.feature_extractor is not None:
+            keys.append("data")
+        if self.targets is not None:
+            keys.append("classes")
+        return [*keys, "orth", "raw"]
+
+    def get_data_shape(self, key: str):
         """
         :returns get_data(*, key).shape[1:], i.e. num-frames excluded
         :rtype: list[int]
         """
+        assert key in self.get_data_keys()
```

Comment:
Not necessary. Instead, add the missing `else:` branch below.

Reply:
Why is it not necessary? Do you think it's appropriate to return values here for data keys that don't exist in the instance, e.g. when feature extraction is turned off and `data` does not exist?